pyccel/ast/builtin_methods/dict_methods.py

```diff
@@ -451,3 +452,50 @@ def key(self):
         The key that is used to select the element from the dict.
         """
         return self._args[0]
+
+#==============================================================================
+class DictValues(Iterable):
+    """
+    Represents a call to the .values() method.
+
+    Represents a call to the .values() method which iterates over the keys of a
+    dictionary.
+
+    Parameters
+    ----------
+    dict_obj : TypedAstNode
+        The object from which the method is called.
+    """
+    __slots__ = ('_dict_obj',)
+    _attribute_nodes = Iterable._attribute_nodes + ("_dict_obj",)
+    _shape = None
+    _class_type = SymbolicType()
+    name = 'keys'
+
+    def __init__(self, dict_obj):
+        self._dict_obj = dict_obj
+        super().__init__(1)
+
+    @property
+    def variable(self):
+        """
+        Get the object representing the dict.
+
+        Get the object representing the dict.
+        """
+        return self._dict_obj
+
+    def get_python_iterable_item(self):
+        """
+        Get the item of the iterable that will be saved to the loop targets.
+
+        Returns two objects that could be a key and a value of the dictionary.
+        These elements are used to determine the types of the loop targets.
+
+        Returns
+        -------
+        list[TypedAstNode]
+            A list of objects that should be assigned to variables.
+        """
+        item = DictPopitem(self._dict_obj)
+        return [IndexedElement(item, 1)]
```

Comment:
Why do we use `DictPopitem` here? `d.popitem()` modifies the dictionary `d` in place, while `d.values()` does not.

Reply:
This method returns an object with the type of the target. It is used in the semantic stage. Eg if we have:
```python
for v in d.values():
```
The type of `v` is deduced by parsing:
```python
v = d.popitem()[1]
```

I didn't want to create a new object to handle this as it is only used during the assignment. However now I think about it again, this might not be the correct solution. For multi dimensional objects this will result in `v` being a heap object when it should be an alias

Reply:
Gotcha. How do you want to proceed?

Reply:
I can probably replace the call to DictPopitem with a Variable. I'll have a look on Monday or Tuesday

Reply:
Done : 915968b20

Reply:
This should be good enough for this PR though. If it is wrong then the fix is simply heap->alias